js/navigation.js

```diff
@@ -1,77 +1,207 @@
+
+	const activateSubmenu = ($button, afterAnimationCallback) => {
+		const animationOptions = isSlideNavigationVisible() ? { right: '0px' } : {};
+		const animationSpeed = isSlideNavigationVisible() ? 250 : 0;
+		$button
+			.attr('aria-expanded', true)
+			.closest('li')
+			.addClass('active')
+			.find('.submenu-wrapper')
+			.animate(animationOptions, animationSpeed, function afterAnimation() {
+				$(this)
+					.find('ul')
+					.attr('aria-hidden', false);
+
+				if (afterAnimationCallback && typeof afterAnimationCallback === 'function') {
+					afterAnimationCallback();
+				}
+			});
 	};
 
-	const removeActiveClassFromAllButtons = ($button) => $button.closest('ul').find('li.active').removeClass('active');
-
-	const toggleActiveClass = ($button) => $button.closest('li').toggleClass('active');
-
-	const removeActiveClass = ($buttonOrList) => $buttonOrList.closest('.active').removeClass('active');
-
-	const hideSearchBox = () => {
-		$('#activate-search-button, #search-box').removeClass('active');
+	const deactivateSubmenu = ($button, afterAnimationCallback) => {
+		const animationOptions = isSlideNavigationVisible() ? { right: '-300px' } : {};
+		const animationSpeed = isSlideNavigationVisible() ? 250 : 0;
+		$button
+			.siblings('.submenu-wrapper')
+			.animate(animationOptions, animationSpeed, function afterAnimation() {
+				$(this)
+					.siblings('button')
+					.attr('aria-expanded', false)
+					.closest('li')
+					.removeClass('active')
+					.attr('aria-hidden', true);
+
+				if (afterAnimationCallback && typeof afterAnimationCallback === 'function') {
+					afterAnimationCallback();
+				}
+			});
 	};
 
-	const equalizeListItems = ($childOfTargetList) => {
-		const $wideList = $childOfTargetList.siblings('ul');
-		const $listItems = $wideList.find('li');
-		let widest = 0;
-		let tallest = 0;
-
-		if ($listItems.length < 8) return;
+	const removeActiveClassFromAllButtons = () => deactivateSubmenu($('#responsive-sliding-navigation').find(activeMenuButtonSelector));
 
-		$listItems.each((listItemIndex, listItem) => {
-			const $listItem = $(listItem);
-			widest = $listItem.width() > widest ? $listItem.width() : widest;
-			tallest = $listItem.height() > tallest ? $listItem.height() : tallest;
-		});
+	const hideSearchBox = () => $(searchArtifactsSelector).removeClass('active');
 
-		$listItems.width(widest);
-		$listItems.height(tallest);
-		$wideList.addClass('wide');
+	const hideHeroCallout = (shouldHide) => {
+		if (shouldHide && !isMobileWidth()) {
+			$(heroCalloutContainerSelector).hide();
+		} else {
+			$(heroCalloutContainerSelector).show();
+		}
 	};
 
-	const navButtonKeyup = (event) => {
+	const navButtonClicked = (event) => {
 		const $button = $(event.currentTarget);
-		const keyCode = event.which || event.keyCode;
+		const isActive = $button.closest('li').hasClass('active');
+		hideSearchBox();
+		removeActiveClassFromAllButtons();
+		if (!isActive) {
+			activateSubmenu($button);
+			hideHeroCallout(true);
```

Comment:
move this after the if and call passing in !isActive

Reply:
Changed the variable to wasActive to be clearer as to my intention. Since activateSubmenu will add the active class back to the correct parent, I can't move the call to `removeActiveClassFromAllButtons();`.